src/cfnlint/data/AdditionalSpecs/SubNeededExcludes.json

```
@@ -169,5 +58,45 @@
                 }
             }
         }
+    },
+    "GlobalPropertyTypes": {
+        "PolicyDocument": {
+            "ExcludeValues": [
+                "aws:CurrentTime",
+                "aws:EpochTime",
+                "aws:TokenIssueTime",
+                "aws:principaltype",
+                "aws:SecureTransport",
+                "aws:SourceIp",
+                "aws:UserAgent",
+                "aws:userid",
+                "aws:username",
+                "ec2:SourceInstanceARN",
+                "iot:Connection.Thing.ThingName",
+                "iot:Connection.Thing.ThingTypeName",
+                "iot:Connection.Thing.IsAttached",
+                "iot:ClientId",
+                "transfer:HomeBucket",
+                "transfer:HomeDirectory",
+                "transfer:HomeFolder",
+                "transfer:UserName",
+                "redshift:DbUser",
+                "cognito-identity.amazonaws.com:aud",
+                "cognito-identity.amazonaws.com:sub",
+                "cognito-identity.amazonaws.com:amr"
+            ]
+        },
+        "UserData": {
+            "ExcludeAll": true
+        },
+        "RequestMappingTemplate": {
+            "ExcludeAll": true
+        },
+        "ResponseMappingTemplate": {
+            "ExcludeAll": true
+        },
+        "CloudWatchAlarmDefinition": {
+            "ExcludeAll": true
+        }
```

IAM Policy conditions are meant to be here too.

```suggestion
        },
        "Condition": {
            "ExcludeAll": true
        }
```

---

Are IAM policy conditions not already matched by the `PolicyDocument` block on Line 63?

---

Good question.  `Condition` was a really early add so I wonder if it still even applies.

---

It used to be `Resource`, `NotResource`, and `Condition`, however #1662 shows that this wasn't sufficient as policies can be embedded as a JSON string in the `PolicyDocument` property. I merged the three and put it at the `PolicyDocument` level, which will also match all sub-properties of a `PolicyDocument` if expressed as a JSON or YAML document rather than an embedded string.